### PR TITLE
chore(component-testing): address comments from TR review

### PR DESCRIPTION
**Pull request by lmiller1990**
Addresses most of the comments from the feat/component-testing review. Those can be [found here](https://cypress-io.atlassian.net/browse/CT-257?focusedCommentId=13070).

Some further reduction of duplicate can be done, but we need something like `shared/server` and `shared/runner` for that to be done in a non-bad manner.

### CODE REVIEW

File: npm/eslint-plugin-dev/lib/index.js [hunk 1]
```diff
@@ -300,43 +300,17 @@ module.exports = {
...
-          react: {
-            version: 'detect', // Tells eslint-plugin-react to automatically detect the version of React to use
-          },
-        },
         plugins: [
           '@typescript-eslint',
           'react',
         ],
         rules: {
           'no-unused-vars': 'off', // avoid interface imports to be warned against
```
Comment (lmiller1990) [inline npm/eslint-plugin-dev/lib/index.js:310]: For some reason we have some linting problems with `spec.tsx` files - there are no other `spec.tsx` files in the code-base. I think we can fix this by updating the version of eslint and the other deps, but a lot of things broke when I tried, so I gave up and just left it as this for now.

File: npm/vue/src/index.ts [hunk 2]
```diff
@@ -14,14 +14,6 @@ const defaultOptions: (keyof MountOptions)[] = [
   'stylesheets',
 ]
 
-function checkMountModeEnabled () {
```
Comment (lmiller1990) [inline npm/vue/src/index.ts:17]: Now we have CT runner, we can assume if the user is using `mount` they are in CT mode.

File: packages/runner-ct/src/.eslintrc.json [hunk 3]
```diff
@@ -1,26 +1,5 @@
...
-        "extensions": [
-          ".js",
-          ".jsx",
-          ".tsx"
-        ]
-      }
-    ]
-  }
+    "../../reporter/src/.eslintrc.json"
+  ]
```
Comment (lmiller1990) [inline packages/runner-ct/src/.eslintrc.json:4]: Probably need a better place to have base eslintrc files... basically every package has copy-pasted the same one and made minor changes.

**Review by flotwig — APPROVED** [review]
LGTM!

**Comment by lmiller1990** [comment]
Closed in favor of: https://github.com/cypress-io/cypress/pull/14966

**Pull request closed without merging**